bin/vault_sync_kv_credential.py

```diff
@@ -134,9 +141,15 @@ def stream_events(self, inputs, ew):
         # logging can't be configured until this point
         # there is no session key available during __init__ (or get_scheme), and we need it to get the running config
         self.configure_logging()
-
         self._logger.debug("stream_events")
 
+        try:
+            self._stream_events(inputs, ew)
+        except Exception as e:
+            self._logger.critical("unhandled exception: {0}".format(traceback.format_exc()))
+            exit(-1)
+
+    def _stream_events(self, inputs, ew):
```

Comment:
I made `stream_events` minimal, doing most of the work in `_stream_events`, so we can catch unhandled exceptions and actually log them sanely.  Without this, the SDK ends up logging an unhandled exception in `splunkd.log` and with only the stringied value of the exception, which seems to always be a single word and entirely unhelpful.